### PR TITLE
Provide `zero-copy` cargo feature

### PR DESCRIPTION
**Pull request by temeddix**
This pull request was inspired by #38

### CODE REVIEW

**Comment by temeddix** [comment]
If there's anything I can do please let me know :)

File: Cargo.toml [hunk 1]
```diff
@@ -36,6 +36,7 @@ required-features = ["anyhow", "backtrace", "chrono", "uuid"]
 [features]
 default = []
 catch-unwind = ["pin-project"]
+prefer-zero-copy = []
```
Comment (shekohex) [inline Cargo.toml:39]: lets just go with `zero-copy` here.
```suggestion
zero-copy = []
```

File: src/into_dart.rs [hunk 2]
```diff
@@ -165,6 +165,7 @@ macro_rules! dart_typed_data_type_trait_impl {
                 }
             }
 
+            #[cfg(not(feature="prefer-zero-copy"))]
```
Comment (shekohex) [inline src/into_dart.rs:168]:
```suggestion
            #[cfg(not(feature="zero-copy"))]
```

File: src/into_dart.rs [hunk 3]
```diff
@@ -181,6 +182,29 @@ macro_rules! dart_typed_data_type_trait_impl {
                     }
                 }
             }
+            #[cfg(feature="prefer-zero-copy")]
```
Comment (shekohex) [inline src/into_dart.rs:185]:
```suggestion
            #[cfg(feature="zero-copy")]
```

**Review by shekohex — CHANGES_REQUESTED** [review]
Looks good! just left a few small comments.

I would also like that if you could mention or add a section in the rustdocs in the `lib.rs` that describe each feature and what is expected when they are enabled.

**Comment by temeddix** [comment]
I'm not sure about what `catch-unwind` is, so please check if my explanation written in `lib.rs` is correct

**Review by shekohex — APPROVED** [review]
LGTM! Thank you.

**Comment by temeddix** [comment]
No problem!

**Comment by temeddix** [comment]
I know it's out of my scope, but may I ask you when this will be provided as a new version, possibly `0.1.15`?

**Comment by shekohex** [comment]
> I know it's out of my scope, but may I ask you when this will be provided as a new version, possibly `0.1.15`?

Very soon. Probably tomorrow morning.

**Pull request merged**